CAM-12651: feat(engine): lock external task without fetch

koevskinikola opened the pull request:
* Allow for an External Task to be locked without a fetch operation.

Related to CAM-12651, CAM-7170

yanavasileva commented on engine/src/main/java/org/camunda/bpm/engine/impl/cmd/LockExternalTaskCmd.java:
```diff
@@ -0,0 +1,69 @@
+  @Override
+  protected void validateInput() {
+    super.validateInput();
+    EnsureUtil.ensureNotNull("lockDuration", lockDuration);
+    EnsureUtil.ensurePositive(BadUserRequestException.class, "lockDuration", lockDuration);
```
I checked that ensurePositive already checks for ensure not null, so we can simplify it to:
```suggestion
    EnsureUtil.ensurePositive(BadUserRequestException.class, "lockDuration", lockDuration);
```

koevskinikola commented:
@yanavasileva I implemented the review hints. For the External Task authorization tests, there was a lot of duplicated code, so I refactored the `HandleExternalTaskAuthorizationTest` class into two classes:

1. `HandleExternalTaskAuthorizationTest` that contains the authorization scenarios, setup, and cleanup;
1. `HandleLockedExternalTaskAuthorizationTest` for testing operations on an already locked task.

I also cleaned up some of the test classes that use the same scenarios.
